### PR TITLE
0.12b Operator Tuning Database and related Tuner Changes.

### PR DESCRIPTION
**Pull request by xinyazhang**
# Overview

This PR adds operator-level backend tuning (`flash_op`) to the Tuner V3.5
pipeline and ships updated kernel tuning databases for all supported
architectures.

With `flash_op`, the tuner selects the best backend (e.g. Triton vs AITER)
per operator configuration, complementing the existing per-HSACO kernel tuning.
The operator API now has a complete two-level tuning path: kernel selection
followed by backend selection.

## Major Changes

* [`flash_op`] Add `flash_op` module for operator-level backend tuning
  + `FlashOp` inherits `Flash`; `BACKEND_COUNT` is arch-dependent for fwd and bwd
  + Backend early-reject via `hipErrorPeerAccessUnsupported` to skip
    unsupported backends without full tuning runs; early-rejected entries
    are recorded with `adiff = -217`
  + Fix `dq_acc` allocation, `prepare_directs` unpacking, and `dq`/`db` outputs
    in `attn_bwd_op`
* [tune] Extend the full tuning pipeline to support `--tuning_mode kernel|op`
  + New PG tables: `optune_results`, `most_accurate_optune_results`,
    `best_optune_results`; new views `kernel_queue_progress` / `op_queue_progress`
  + `populate_most_accurate`, `compute_best_results`, `export_best_results`
    extended with `SqlStatements` abstraction covering both modes
  + `retry_broken_entries` / `reset_broken_to_pending` route to correct tables
    based on `--tuning_mode`
  + Note: workers must be restarted to switch tuning mode; kernel-mode and
    op-mode workers load disjoint module sets (`flash` vs `flash_op`)
* [tune] Fix `compute_best_results` ignoring `filter_task_ids` in incremental/fix
  mode: worker queried the full `BETWEEN` band instead of the requested task IDs,
  causing the entire suite to be reprocessed. More critically, only the
  `most_accurate_results` for the requested task IDs were loaded into memory,
  so the reprocessed entries outside that set had no correctness data and fell
  back to the fastest kernel regardless of accuracy — effectively destroying the
  tuning database on any partial update.
  + `compute_best_results` now treats backends that fail `00_*` unit tests
    (the basic correctness checks) as unusable, preventing incorrect backends
    from being selected as best
* [webui] Dashboard, debug views, Bake LUT, and worker controls updated
  throughout to support `tuning_mode kernel|op`
* [db] Update kernel+op tuning databases for all architectures (v3+level3),
  all passing unit tests (see Known Problems for residual failures)

## Minor Changes

* [tune/tdesc] Refactor tuning infrastructure to use general terminology:
  `which_impl` as the unified abstract parameter name across all tuning modules
* [flash/kernels] Split `kernels.py` into `kernel_calls.py` (API call
  sequences) and `kernel_opts.py` (`attn_options` construction). `flash` and
  `flash_op` cannot share `AttnOptionsWrapperOp` because `attn_options` is
  compiled with different members in kernel vs op tuning modes (op tuning uses
  a testing build). Since `attn_options` is the only difference in how the API
  is called, splitting into two hierarchies lets `flash_op` reuse
  `kernel_calls.py` directly.
* [flash/module] Skip gfx11xx entries with `seqlen_q/k > 2048`
* [test] Catch `torch.AcceleratorError` in `_core_test_backward` and exit cleanly
* [test] Fix `SMALL_VRAM` check to account for dtype size; free GPU memory
  proactively for large configs
* [test/adiffs] Add accuracy issue records for gfx90a, gfx950, gfx1100, gfx1201
* [codegen] `sancheck` reports op tuning LUT coverage statistics
* [tune] Add `dbsum` script to sha256sum all sqlite3 databases; checksums
  the uncompressed `.sqlite3` content (same as `__signature__`), not the
  `.tar.xz` archive
* [tune] Add `.tune/remote/bash_direct` to drop into an interactive bash shell inside
  the tuning Docker image with venv and ROCm environment fully configured,
  mirroring `testrun_direct` for easier debugging

## Known Problems

* gfx1100: < 10 of UTs fail due to compiler accuracy issues.
  See `test/adiffs/gfx1100.txt` for details.
* gfx1201: a small number of UTs fail due to hipblasLt GPU segfault, can be
  confirmed with `AOTRITON_REF_DEVICE_OPTION=cpu`.
* gfx950: `hdim=48/80` bwd kernels remain disabled. This is a compiler
  numerical accuracy bug first seen in 0.11 and still not fully fixed per
  our tests; keeping them disabled until confirmed clean.
* Op name inconsistency: tuning stores op names with `_op` suffix (e.g.
  `attn_fwd_op`) while SQLite tables use `op_` prefix (e.g. `op_attn_fwd`);
  to be fixed by unifying to `op_` prefix throughout.

### CODE REVIEW

File: .tune/bin/export_best_results [hunk 1]
```diff
@@ -33,22 +35,35 @@
...
 shift
+TUNING_MODE="kernel"
 
-# Default output: <workdir>/scratch/centraldb.sqlite3
+# Peek ahead to extract --tuning_mode before deciding the default output path.
+for arg in "$@"; do
+    if [ "$arg" = "op" ] && [ "${prev_arg:-}" = "--tuning_mode" ]; then
+        TUNING_MODE="op"
+    fi
+    prev_arg="$arg"
```
[the remark this answers is not in the document]
Comment (xinyazhang) [reply .tune/bin/export_best_results:57]: Invalid, we won't support the --tuning_mode=op syntax.

File: v3python/tune/flash_op/kernels.py [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+        return (devm.dk, devm.dv, devm.dq, devm.db), err
+
+    def prepare_directs(self, im, inputs):
+        im, view, devm = super().prepare_directs(im, inputs)
+        from pyaotriton import lazy_tensor
+        from ..gpu_utils import mk_aotensor
+        devm.dq_acc = torch.zeros(*devm.q.size(), dtype=torch.float32, device=devm.q.device)
+        dq_acc_view, _ = mk_aotensor(devm.dq_acc)
+        view.dq_acc = lazy_tensor.eager_null_dq_acc(dq_acc_view)
+        return im, view, devm
```
[the remark this answers is not in the document]
Comment (xinyazhang) [reply v3python/tune/flash_op/kernels.py:96]: valid but I didn't bother to fix
